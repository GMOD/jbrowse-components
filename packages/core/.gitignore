.DS_Store
node_modules
coverage
dist
*.log
*.swp
*.js
*.d.ts
*.js.map
./BaseFeatureWidget
./ReExports
./TextSearch
./assemblyManager
./configuration
./data_adapterts
./pluggableElementTypes
./rpc
./ui
./util
