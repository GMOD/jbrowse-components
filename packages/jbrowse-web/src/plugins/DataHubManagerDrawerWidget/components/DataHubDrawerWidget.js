import Button from '@material-ui/core/Button'
import Step from '@material-ui/core/Step'
import StepContent from '@material-ui/core/StepContent'
import StepLabel from '@material-ui/core/StepLabel'
import Stepper from '@material-ui/core/Stepper'
import { withStyles } from '@material-ui/core/styles'
import Typography from '@material-ui/core/Typography'
import { observer, PropTypes as MobxPropTypes } from 'mobx-react'
import { getRoot, applySnapshot, getSnapshot } from 'mobx-state-tree'
import propTypes from 'prop-types'
import React from 'react'
import ConfirmationDialog from './ConfirmationDialog'
import HubSourceSelect from './HubSourceSelect'
import HubTypeSelect from './HubTypeSelect'
import TrackHubRegistrySelect from './TrackHubRegistrySelect'
import UrlInput from './UrlInput'

const styles = theme => ({
  root: {
    marginTop: theme.spacing.unit,
  },
  stepper: {
    backgroundColor: theme.palette.background.default,
  },
  button: {
    marginTop: theme.spacing.unit,
    marginRight: theme.spacing.unit,
  },
  actionsContainer: {
    marginBottom: theme.spacing.unit * 2,
  },
})

const steps = [
  'Select a Data Hub Type',
  'Select a Data Hub Source',
  'Select a Data Hub',
  'Confirm Selection',
]

class DataHubDrawerWidget extends React.Component {
  static propTypes = {
    classes: propTypes.shape({
      root: propTypes.string.isRequired,
      stepper: propTypes.string.isRequired,
      button: propTypes.string.isRequired,
      actionsContainer: propTypes.string.isRequired,
    }).isRequired,
    model: MobxPropTypes.observableObject.isRequired,
  }

  state = {
    // Step 0
    hubType: null, // ucsc, jbrowse1
    // Step 1
    hubSource: null, // trackHubRegistry, ucscCustom, jbrowseRegistry, jbrowseCustom
    // Step 2
    trackDbUrl: '',
    hubName: '',
    assemblyName: '',
    // Step 3
    // eslint-disable-next-line react/no-unused-state
    backupRootModel: null,

    activeStep: 0,
    nextEnabledThroughStep: -1,
  }

  setHubType = event => this.setState({ hubType: event.target.value })

  setHubSource = event => this.setState({ hubSource: event.target.value })

  setTrackDbUrl = newUrl => this.setState({ trackDbUrl: newUrl })

  setHubName = newHubName => this.setState({ hubName: newHubName })

  setAssemblyName = newAssemblyName =>
    this.setState({ assemblyName: newAssemblyName })

  get stepContent() {
    const {
      activeStep,
      hubType,
      hubSource,
      trackDbUrl,
      hubName,
      assemblyName,
    } = this.state
    const { model } = this.props
    let StepComponent
    switch (activeStep) {
      case 0:
        return (
          <HubTypeSelect
            hubType={hubType}
            setHubType={this.setHubType}
            enableNext={() => this.enableNextThrough(activeStep)}
          />
        )
      case 1:
        return (
          <HubSourceSelect
            hubType={hubType}
            hubSource={hubSource}
            setHubSource={this.setHubSource}
            enableNext={() => this.enableNextThrough(activeStep)}
          />
        )
      case 2:
        if (hubSource === 'ucscCustom') StepComponent = UrlInput
        else if (hubSource === 'trackHubRegistry')
          StepComponent = TrackHubRegistrySelect
        else
          return <Typography color="error">Unknown Data Hub Source</Typography>
        return (
          <StepComponent
            enableNext={() => this.enableNextThrough(activeStep)}
            disableNext={() => this.enableNextThrough(activeStep - 1)}
            setTrackDbUrl={this.setTrackDbUrl}
            setHubName={this.setHubName}
            setAssemblyName={this.setAssemblyName}
          />
        )
      case 3:
        return (
          <ConfirmationDialog
            hubName={hubName}
            assemblyName={assemblyName}
            trackDbUrl={trackDbUrl}
            enableNext={() => this.enableNextThrough(activeStep)}
            rootModel={getRoot(model)}
          />
        )
      default:
        return <Typography>Unknown step</Typography>
    }
  }

  enableNextThrough = step => this.setState({ nextEnabledThroughStep: step })

  handleNext = () => {
    const { activeStep } = this.state
    const { model } = this.props
    const rootModel = getRoot(model)
    if (activeStep === steps.length - 1) {
      rootModel.hideAllDrawerWidgets()
      return
    }
    if (activeStep === steps.length - 2) {
      const backupRootModel = getSnapshot(rootModel)
      // eslint-disable-next-line react/no-unused-state
      this.setState({ backupRootModel })
    }
    this.setState(state => ({
      activeStep: state.activeStep + 1,
    }))
  }

  handleBack = () => {
    this.setState((state, props) => {
      let { hubSource, backupRootModel } = state
      const { rootModel } = props
      const { activeStep } = state
      if (activeStep === steps.length - 1) {
        applySnapshot(rootModel, backupRootModel)
        backupRootModel = null
      }
      const newStep = activeStep - 1
      if (newStep < 1) hubSource = null
      return {
        activeStep: newStep,
        nextEnabledThroughStep: newStep,
        hubSource,
        backupRootModel,
      }
    })
  }

  render() {
    const { classes } = this.props
    const { activeStep, nextEnabledThroughStep } = this.state

    return (
      <div className={classes.root}>
        <Stepper
          className={classes.stepper}
          activeStep={activeStep}
          orientation="vertical"
        >
          {steps.map((label, index) => (
            <Step key={label}>
              <StepLabel>{label}</StepLabel>
              <StepContent>
                {this.stepContent}
                <div className={classes.actionsContainer}>
                  <Button
                    disabled={activeStep === 0}
                    onClick={this.handleBack}
                    className={classes.button}
                  >
                    Back
                  </Button>
                  <Button
                    disabled={index > nextEnabledThroughStep}
                    variant="contained"
                    color="primary"
                    onClick={this.handleNext}
                    className={classes.button}
                  >
                    {activeStep === steps.length - 1 ? 'Finish' : 'Next'}
                  </Button>
                </div>
              </StepContent>
            </Step>
          ))}
        </Stepper>
      </div>
    )
  }
}

export default withStyles(styles)(observer(DataHubDrawerWidget))
