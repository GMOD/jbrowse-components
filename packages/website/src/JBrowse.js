import { readConfObject } from '@gmod/jbrowse-core/configuration'
import { App, theme } from '@gmod/jbrowse-core/ui'
import { toUrlSafeB64 } from '@gmod/jbrowse-core/util'
import CssBaseline from '@material-ui/core/CssBaseline'
import { ThemeProvider } from '@material-ui/core/styles'
import { observer } from 'mobx-react'
import { getSnapshot, onSnapshot } from 'mobx-state-tree'
import { StringParam, useQueryParam } from 'use-query-params'
import React, { useEffect } from 'react'

const MAX_SESSION_SIZE_IN_URL = 100000

// adapted from https://github.com/jashkenas/underscore/blob/5d8ab5e37c9724f6f1181c5f95d0020815e4cb77/underscore.js#L894-L925
function debounce(func, wait) {
  let timeout
  let result
  const later = (...args) => {
    timeout = null
    result = func(...args)
  }
  const debounced = (...args) => {
    if (timeout) {
      clearTimeout(timeout)
    }
    timeout = setTimeout(() => {
      return later(...args)
    }, wait)
    return result
  }
  debounced.cancel = () => {
    clearTimeout(timeout)
    timeout = null
  }
  return debounced
}

const JBrowse = observer(({ pluginManager }) => {
  const [, setSession] = useQueryParam('session', StringParam)

  const { rootModel } = pluginManager
  const { session, jbrowse, error } = rootModel || {}
  const useLocalStorage = jbrowse
    ? readConfObject(jbrowse.configuration, 'useLocalStorage')
    : false

  const useUpdateUrl = jbrowse
    ? readConfObject(jbrowse.configuration, 'useUrlSession')
    : false

  // Set session URL on first render only, before `onSnapshot` has fired
  useEffect(() => {
    if (useUpdateUrl) {
      const json = JSON.stringify(getSnapshot(session))
      const sess =
        json.length < MAX_SESSION_SIZE_IN_URL ? toUrlSafeB64(json) : undefined
      setSession(sess)
    }
  }, [session, setSession, useUpdateUrl])

  useEffect(() => {
    function updateUrl(snapshot) {
      const json = JSON.stringify(snapshot)
      const sess =
        json.length < MAX_SESSION_SIZE_IN_URL ? toUrlSafeB64(json) : undefined

      setSession(sess)
      if (rootModel && rootModel.session) {
        rootModel.jbrowse.updateSavedSession(snapshot)
      }
      if (useLocalStorage) {
        localStorage.setItem('jbrowse-web-session', JSON.stringify(snapshot))
      }
    }

    let disposer = () => {}
    if (session && useUpdateUrl) {
      const updater = debounce(updateUrl, 400)
      const snapshotDisposer = onSnapshot(session, updater)
      disposer = () => {
        snapshotDisposer()
        updater.cancel()
      }
    }
    return disposer
  }, [rootModel, setSession, useLocalStorage, useUpdateUrl, session])

  if (error) {
    throw new Error(error)
  }

  return <App session={rootModel.session} />
})

export default props => {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <JBrowse {...props} />
    </ThemeProvider>
  )
}
