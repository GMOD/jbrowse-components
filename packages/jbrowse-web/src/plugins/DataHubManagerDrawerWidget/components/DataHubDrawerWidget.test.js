import { createShallow, getClasses } from '@material-ui/core/test-utils'
import React from 'react'
import JBrowse from '../../../JBrowse'
import DataHubDrawerWidget from './DataHubDrawerWidget'

describe('<DataHubDrawerWidget />', () => {
  let shallow
  let classes
  let jbrowse
  let rootModel

  beforeAll(async () => {
    shallow = createShallow({ untilSelector: 'DataHubDrawerWidget' })
    classes = getClasses(<DataHubDrawerWidget rootModel={rootModel} />)
    jbrowse = await new JBrowse().configure({ configId: 'testing' })
    rootModel = jbrowse.model
  })

  it('renders', () => {
    const wrapper = shallow(<DataHubDrawerWidget rootModel={rootModel} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.hasClass(classes.root)).toBe(true)
    expect(wrapper.find('WithStyles(Stepper)').hasClass(classes.stepper)).toBe(
      true,
    )
    expect(
      wrapper
        .find('WithStyles(Button)')
        .at(0)
        .hasClass(classes.button),
    ).toBe(true)
    expect(
      wrapper
        .find(`div.${classes.actionsContainer}`)
        .at(0)
        .hasClass(classes.actionsContainer),
    ).toBe(true)
    expect(
      wrapper
        .find(`div.${classes.actionsContainer}`)
        .at(0)
        .hasClass(classes.actionsContainer),
    ).toBe(true)
  })

  it('can handle a custom UCSC URL', () => {
    const wrapper = shallow(<DataHubDrawerWidget rootModel={rootModel} />)
    expect(wrapper).toMatchSnapshot()
    const instance = wrapper.instance()
    instance.setHubType({ target: { value: 'ucsc' } })
    instance.enableNextThrough(0)
    instance.handleNext()
    instance.handleBack()
    instance.handleNext()
    instance.setHubSource({ target: { value: 'ucscCustom' } })
    instance.enableNextThrough(1)
    instance.handleNext()
    instance.setTrackDbUrl(new URL('http://www.example.com/trackDb.txt'))
    instance.setHubName('testHubName')
    instance.setAssemblyName('testAssemblyName')
    instance.enableNextThrough(2)
    instance.handleNext()
    instance.enableNextThrough(3)
    instance.handleNext()
    expect(wrapper).toMatchSnapshot()
  })

  it('can handle a custom UCSC URL', () => {
    const wrapper = shallow(<DataHubDrawerWidget rootModel={rootModel} />)
    expect(wrapper).toMatchSnapshot()
    const instance = wrapper.instance()
    instance.setHubType({ target: { value: 'ucsc' } })
    instance.enableNextThrough(0)
    instance.handleNext()
    instance.handleBack()
    instance.handleNext()
    instance.setHubSource({ target: { value: 'trackHubRegistry' } })
    instance.enableNextThrough(1)
    instance.handleNext()
    instance.setTrackDbUrl(new URL('http://www.example.com/trackDb.txt'))
    instance.setHubName('testHubName')
    instance.setAssemblyName('testAssemblyName')
    instance.enableNextThrough(2)
    instance.handleNext()
    instance.enableNextThrough(3)
    instance.handleNext()
    expect(wrapper).toMatchSnapshot()
  })
})
