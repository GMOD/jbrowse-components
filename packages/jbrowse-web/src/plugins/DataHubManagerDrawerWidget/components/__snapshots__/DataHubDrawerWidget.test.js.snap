// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataHubDrawerWidget /> can handle a custom UCSC URL 1`] = `
<div
  className="inject-DataHubDrawerWidget-with-rootModel-root-1"
>
  <WithStyles(Stepper)
    activeStep={0}
    className="inject-DataHubDrawerWidget-with-rootModel-stepper-2"
    orientation="vertical"
  >
    <WithStyles(Step)
      key="Select a Data Hub Type"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Type
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub Source"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Source
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Confirm Selection"
    >
      <WithStyles(StepLabel)>
        Confirm Selection
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
  </WithStyles(Stepper)>
</div>
`;

exports[`<DataHubDrawerWidget /> can handle a custom UCSC URL 2`] = `
<div
  className="inject-DataHubDrawerWidget-with-rootModel-root-1"
>
  <WithStyles(Stepper)
    activeStep={3}
    className="inject-DataHubDrawerWidget-with-rootModel-stepper-2"
    orientation="vertical"
  >
    <WithStyles(Step)
      key="Select a Data Hub Type"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Type
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <WithStyles(inject-ConfirmationDialog-with-rootModel)
          assemblyName="testAssemblyName"
          enableNext={[Function]}
          hubName="testHubName"
          trackDbUrl={"http://www.example.com/trackDb.txt"}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={false}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={false}
            onClick={[Function]}
            variant="contained"
          >
            Finish
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub Source"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Source
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <WithStyles(inject-ConfirmationDialog-with-rootModel)
          assemblyName="testAssemblyName"
          enableNext={[Function]}
          hubName="testHubName"
          trackDbUrl={"http://www.example.com/trackDb.txt"}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={false}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={false}
            onClick={[Function]}
            variant="contained"
          >
            Finish
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <WithStyles(inject-ConfirmationDialog-with-rootModel)
          assemblyName="testAssemblyName"
          enableNext={[Function]}
          hubName="testHubName"
          trackDbUrl={"http://www.example.com/trackDb.txt"}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={false}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={false}
            onClick={[Function]}
            variant="contained"
          >
            Finish
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Confirm Selection"
    >
      <WithStyles(StepLabel)>
        Confirm Selection
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <WithStyles(inject-ConfirmationDialog-with-rootModel)
          assemblyName="testAssemblyName"
          enableNext={[Function]}
          hubName="testHubName"
          trackDbUrl={"http://www.example.com/trackDb.txt"}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={false}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={false}
            onClick={[Function]}
            variant="contained"
          >
            Finish
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
  </WithStyles(Stepper)>
</div>
`;

exports[`<DataHubDrawerWidget /> can handle a custom UCSC URL 3`] = `
<div
  className="inject-DataHubDrawerWidget-with-rootModel-root-1"
>
  <WithStyles(Stepper)
    activeStep={0}
    className="inject-DataHubDrawerWidget-with-rootModel-stepper-2"
    orientation="vertical"
  >
    <WithStyles(Step)
      key="Select a Data Hub Type"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Type
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub Source"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Source
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Confirm Selection"
    >
      <WithStyles(StepLabel)>
        Confirm Selection
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
  </WithStyles(Stepper)>
</div>
`;

exports[`<DataHubDrawerWidget /> can handle a custom UCSC URL 4`] = `
<div
  className="inject-DataHubDrawerWidget-with-rootModel-root-1"
>
  <WithStyles(Stepper)
    activeStep={3}
    className="inject-DataHubDrawerWidget-with-rootModel-stepper-2"
    orientation="vertical"
  >
    <WithStyles(Step)
      key="Select a Data Hub Type"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Type
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <WithStyles(inject-ConfirmationDialog-with-rootModel)
          assemblyName="testAssemblyName"
          enableNext={[Function]}
          hubName="testHubName"
          trackDbUrl={"http://www.example.com/trackDb.txt"}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={false}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={false}
            onClick={[Function]}
            variant="contained"
          >
            Finish
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub Source"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Source
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <WithStyles(inject-ConfirmationDialog-with-rootModel)
          assemblyName="testAssemblyName"
          enableNext={[Function]}
          hubName="testHubName"
          trackDbUrl={"http://www.example.com/trackDb.txt"}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={false}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={false}
            onClick={[Function]}
            variant="contained"
          >
            Finish
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <WithStyles(inject-ConfirmationDialog-with-rootModel)
          assemblyName="testAssemblyName"
          enableNext={[Function]}
          hubName="testHubName"
          trackDbUrl={"http://www.example.com/trackDb.txt"}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={false}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={false}
            onClick={[Function]}
            variant="contained"
          >
            Finish
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Confirm Selection"
    >
      <WithStyles(StepLabel)>
        Confirm Selection
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <WithStyles(inject-ConfirmationDialog-with-rootModel)
          assemblyName="testAssemblyName"
          enableNext={[Function]}
          hubName="testHubName"
          trackDbUrl={"http://www.example.com/trackDb.txt"}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={false}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={false}
            onClick={[Function]}
            variant="contained"
          >
            Finish
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
  </WithStyles(Stepper)>
</div>
`;

exports[`<DataHubDrawerWidget /> renders 1`] = `
<div
  className="inject-DataHubDrawerWidget-with-rootModel-root-1"
>
  <WithStyles(Stepper)
    activeStep={0}
    className="inject-DataHubDrawerWidget-with-rootModel-stepper-2"
    orientation="vertical"
  >
    <WithStyles(Step)
      key="Select a Data Hub Type"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Type
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub Source"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub Source
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Select a Data Hub"
    >
      <WithStyles(StepLabel)>
        Select a Data Hub
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
    <WithStyles(Step)
      key="Confirm Selection"
    >
      <WithStyles(StepLabel)>
        Confirm Selection
      </WithStyles(StepLabel)>
      <WithStyles(StepContent)>
        <HubTypeSelect
          enableNext={[Function]}
          hubType={null}
          setHubType={[Function]}
        />
        <div
          className="inject-DataHubDrawerWidget-with-rootModel-actionsContainer-4"
        >
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            disabled={true}
            onClick={[Function]}
          >
            Back
          </WithStyles(Button)>
          <WithStyles(Button)
            className="inject-DataHubDrawerWidget-with-rootModel-button-3"
            color="primary"
            disabled={true}
            onClick={[Function]}
            variant="contained"
          >
            Next
          </WithStyles(Button)>
        </div>
      </WithStyles(StepContent)>
    </WithStyles(Step)>
  </WithStyles(Stepper)>
</div>
`;
